smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ServiceClientGenerator.kt

```diff
@@ -103,45 +137,25 @@ class ServiceGenerator(private val ctx: RenderingContext<ServiceShape>) {
      *
      * e.g.
      * ```
-     * companion object {
-     *     operator fun invoke(block: Config.Builder.() -> Unit = {}): LambdaClient {
-     *         val config = Config.Builder().apply(block).build()
-     *         return DefaultLambdaClient(config)
-     *     }
-     *
-     *     operator fun invoke(config: Config): LambdaClient = DefaultLambdaClient(config)
+     * companion object : SdkClientFactory<Config, Config.Builder, LambdaClient, Builder> {
+     *     override fun builder: Builder = Builder()
      * }
      * ```
      */
     private fun renderCompanionObject() {
-        writer.withBlock("public companion object {", "}") {
-            val hasProtocolGenerator = ctx.protocolGenerator != null
-            // If there is no ProtocolGenerator, do not codegen references to the non-existent default client.
-            callIf(hasProtocolGenerator) {
-                withBlock("public operator fun invoke(block: Config.Builder.() -> Unit = {}): ${serviceSymbol.name} {", "}") {
-                    write("val config = Config.Builder().apply(block).build()")
-                    write("return Default${serviceSymbol.name}(config)")
-                }
-                write("")
-                write("public operator fun invoke(config: Config): ${serviceSymbol.name} = Default${serviceSymbol.name}(config)")
-            }
+        // don't render a companion object which is used for building a service client unless we have a protocol generator
+        if (ctx.protocolGenerator == null) return
+        writer.withBlock(
+            "public companion object : #T<Config, Config.Builder, #T, Builder> {",
+            "}",
+            RuntimeTypes.Core.Client.SdkClientFactory,
+            serviceSymbol,
+        ) {
+            write("@#T", KotlinTypes.Jvm.JvmStatic)
+            write("override fun builder(): Builder = Builder()")
         }
     }
 
-    private fun importExternalSymbols() {
-        // base client interface
-        val sdkInterfaceSymbol = Symbol.builder()
-            .name("SdkClient")
-            .namespace(RUNTIME_ROOT_NS, ".")
-            .addDependency(KotlinDependency.CORE)
-            .build()
-
-        writer.addImport(sdkInterfaceSymbol)
-
-        // import all the models generated for use in input/output shapes
-        writer.addImport("${ctx.settings.pkg.name}.model", "*")
-    }
-
```

Comment:
Hmm, streaming blob types aren't from the `model` package anyway though. I'll take a look.

Reply:
I believe it's because the client interface's DSL overloads use `#T` templates and automatically import their types but the streaming blob types don't get DSL overloads...they're only added during `renderOperation` which uses `OperationIndex.operationSignature` to get the _string_ for method signatures and writes it as a literal (ignoring type imports). This was fine before when we imported `model.*` but now it's broke.

Reply:
Yup I found it and see it now, I think fixing this locally where we generate the operation is most appropriate but if I play too much whack-a-mole I may revert and keep the star import to be tackled another day